src/extractor/guidance/intersection_handler.cpp

```diff
@@ -383,136 +393,140 @@ std::size_t IntersectionHandler::findObviousTurn(const EdgeID via_edge,
-    {
-        const double deviation = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
-        if (!intersection[i].entry_allowed)
-            continue;
+    std::vector<out_way> out_ways;
 
-        const auto out_data = node_based_graph.GetEdgeData(intersection[i].eid);
-        const auto continue_class =
-            node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification;
+    // create sortable vector of only entry allowed out_ways
+    // get a count of number of ways from that intersection that qualify to have
+    // continue instruction because they share a name with the approaching way
+    std::int64_t continue_count = 0, continue_count_valid = 0;
+    for (std::size_t i = 1, last = intersection.size(); i < last; ++i)
+    {
+        out_way out;
+        const auto out_node = node_based_graph.GetEdgeData(intersection[i].eid);
 
-        const auto same_name = !util::guidance::requiresNameAnnounced(
-            in_data.name_id, out_data.name_id, name_table, street_name_suffix_table);
+        out.road = &intersection[i];
+        out.same_name_id = !util::guidance::requiresNameAnnounced(
+            in_way.name_id, out_node.name_id, name_table, street_name_suffix_table);
+        if (out.same_name_id)
+            ++continue_count;
 
-        if (same_name && (best_continue == 0 || (continue_class.GetPriority() >
-                                                     out_data.road_classification.GetPriority() &&
-                                                 in_classification != continue_class) ||
-                          (deviation < best_continue_deviation &&
-                           out_data.road_classification == continue_class) ||
-                          (continue_class != in_classification &&
-                           out_data.road_classification == continue_class)))
+        if (!out.road->entry_allowed)
         {
-            best_continue_deviation = deviation;
-            best_continue = i;
+            ++continue_count_valid;
+            continue;
         }
 
-        const auto current_best_class =
-            node_based_graph.GetEdgeData(intersection[best_continue].eid).road_classification;
+        out.index = i;
+        out.deviation_from_straight = angularDeviation(intersection[i].angle, STRAIGHT_ANGLE);
+        out.same_classification = out_node.road_classification == in_way.road_classification;
+        out.same_or_higher_priority =
+            out_node.road_classification.GetPriority() <= in_way.road_classification.GetPriority();
 
-        // don't prefer low priority classes
-        if (best != 0 && out_data.road_classification.IsLowPriorityRoadClass() &&
-            !current_best_class.IsLowPriorityRoadClass())
-            continue;
+        out_ways.push_back(out);
+    }
+    const std::pair<std::int64_t, std::int64_t> num_continue_names =
+        std::make_pair(continue_count, continue_count_valid);
```

Comment:
see above: this is `continue_count_invalid`. Also, what is the benefit in storing this in a pair? `num_continue_name.first/.second` is far less intuitive then a named variable.